pca-main-nokendra.template

```diff
@@ -589,18 +677,51 @@ Resources:
           - ShouldEnableKendraSearch
           - 'None'
           - 'None'
-        WebUri: !GetAtt PCAUI.Outputs.WebUri
+        # WebUri: !GetAtt PCAUI.Outputs.WebUri
         DatabaseName: !Ref DatabaseName
 
+  PythonUtilsLayer:
+    DependsOn: SSM
+    Type: AWS::CloudFormation::Stack
+    Properties:
+      TemplateURL: pca-server/cfn/lib/python-utilities.template
+
+  BedrockBoto3Layer:
+    Type: AWS::CloudFormation::Stack
+    Condition: ShouldDeployBedrockBoto3Layer
+    Properties:
+      TemplateURL: pca-boto3-bedrock/template.yaml
+
+  LLMThirdPartyApiKeySecret:
+    Type: AWS::SecretsManager::Secret
+    Properties: 
+      Description: API Key
+      Name: !Sub '${AWS::StackName}-ThirdPartyApiKey'
+      SecretString: !Ref SummarizationLLMThirdPartyApiKey
+    Metadata:
+      cfn_nag:
+        rules_to_suppress:
+          - id: W77
+            reason: No requirement for custom CMK, secret will not be shared cross-account.
 
   PCAServer:
     Type: AWS::CloudFormation::Stack
-    DependsOn: SSM
+    # DependsOn: SSM
```

Comment:
Why remove the DependsOn.. this makes me a tad nervous.. doesn't it make a race condition between SSM and PCA stack (that reads the parameters) with stack params like
```
  DatabaseName:
    Type: AWS::SSM::Parameter::Value<String>
    Default: DatabaseName
```